### PR TITLE
Stats - Add the published module

### PR DESCRIPTION
**Pull request by daniloercoli**
#2494 Add the "Published" module to Stats. 

The module is designed to be in par with the web version.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/StatsPublishedPostsFragment.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        } else {
+            showHideNoResultsUI(true);
+        }
+    }
+
+    private boolean hasPublishedPostsAndPages() {
+        return !isDataEmpty() && ((PublishedPostsAndPagesModel) mDatamodels[0]).hasPublishedPostsAndPages();
+    }
+
+    private List<PostModel> getTopPostsAndPages() {
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/StatsPublishedPostsFragment.java]: Should this be renamed to something like mPublishedPostsAndPages?

File: WordPress/src/main/java/org/wordpress/android/ui/stats/StatsPublishedPostsFragment.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    public String getTitle() {
+        return getString(R.string.stats_view_published);
+    }
+
+
+    private class PublishedPostsAndPagesAdapter extends ArrayAdapter<PostModel> {
+
+        private List<PostModel> list;
+        private final LayoutInflater inflater;
+        private final int localTableBlogID;
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/StatsPublishedPostsFragment.java]: This field is unused. Will it be needed in the future or is it worth taking out?

File: WordPress/src/main/java/org/wordpress/android/ui/stats/models/PublishedPostsAndPagesModel.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package org.wordpress.android.ui.stats.models;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.JSONUtils;
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/models/PublishedPostsAndPagesModel.java]: This import and Iterator can be removed.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/models/PublishedPostsAndPagesModel.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+
+public class PublishedPostsAndPagesModel implements Serializable {
+    private final String mBlogID;
+    private List<PostModel> mTopPostsAndPages;
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/models/PublishedPostsAndPagesModel.java]: Should this be renamed to something like mPublishedPostsAndPages?

**Comment by tonyr59h** [comment]
Tested different blogs, different time spans, on Nexus 5 and S3. Looks good and works as expected.

I notice that there's a bit more to the web implementation of the stats views. It's collapsable and has an info icon:
 

Are there plans to bring these views to the mobile platforms?

**Comment by daniloercoli** [comment]
> Are there plans to bring these views to the mobile platforms?

At the beginning of Calypso-Stats we decided to not implement collapsable modules on mobile. And we don't have that feature scheduled in the short term.  I will bring this up during the next Stats squad chat.

**Comment by daniloercoli** [comment]
@tonyr59h - Renamed unused variables and removed unused fields and imports. That weird though, I remember I had already renamed those fields. I probably missed to commit those changes.

**Comment by tonyr59h** [comment]
:shipit:

**Pull request merged**